### PR TITLE
Update License file with information about Redis and Valkey licenses

### PR DESCRIPTION
**Pull request by zuiderkwast**

### CODE REVIEW

**Comment by zuiderkwast** [comment]
I hope this makes sense. We can update it again in the future (like everything else) but this should hopefully be correct enough to publish on a website.

I believe it would be correct to put both versions of the license (with "the copyright holder" and "Redis" in the 3rd clause) in the COPYING file too.

File: topics/license.md [hunk 1]
```diff
@@ -1,25 +1,64 @@
...
-* Redis is **open source software** released under the terms of the **three clause BSD license**. Most of the Redis source code was written and is copyrighted by Salvatore Sanfilippo and Pieter Noordhuis. A list of other contributors can be found in the git history.
+* Valkey is **open source software** released under the terms of the **three clause BSD license**. Most of the Valkey source code was written by Salvatore Sanfilippo and Pieter Noordhuis. A list of other contributors can be found in the git history.
 
-  The Redis trademark and logo are owned by Redis Ltd. and can be
-used in accordance with the [Redis Trademark Guidelines](https://redis.com/legal/trademark-guidelines/).
+* Valkey is based on the formerly open source Redis, as it was before the
+  license of Redis was changed to one that is not open source.
+  Read more about this in the [History of Valkey](history.md).
+  Redis is a trademark of Redis Ltd. Whenever we use the name Redis in the
+  Valkey documentation, we trying our best to use in accordance with the
```
Comment (natoscott) [inline topics/license.md:18]: grammar issue there on "we trying"

File: topics/license.md [hunk 2]
```diff
@@ -1,25 +1,64 @@
...
+  [Redis Trademark Guidelines](https://redis.com/legal/trademark-guidelines/).
 
 ## Licences:
 
 ### Three clause BSD license
 
-Every file in the Redis distribution, with the exceptions of third party files specified in the list below, contain the following license:
+Every file in the Valkey distribution, with the exceptions of third party files
```
Comment (natoscott) [inline topics/license.md:25]: "with the exception" (singular)

File: topics/license.md (hunk 1 above)
Comment (zuiderkwast) [reply topics/license.md:18]:
```suggestion
  Valkey documentation, we're trying our best to use it in accordance with the
```

File: topics/license.md (hunk 2 above)
Comment (zuiderkwast) [reply topics/license.md:25]:
```suggestion
Every file in the Valkey distribution, with the exception of third party files
```

**Review by madolson — APPROVED** [review]

**Pull request merged**